donut/modules/editor/helpers.py

```diff
@@ -0,0 +1,202 @@
+        cursor.execute(query, title)
+        res = cursor.fetchone()
+
+
+def rename_title(oldfilename, newfilename):
+    """
+    Changes the file name of an html file
+    Need to look for paths
+    """
+    oldpath = os.path.join(flask.current_app.root_path,
+                           flask.current_app.config["UPLOAD_WEBPAGES"],
+                           oldfilename + '.md')
+    newpath = os.path.join(flask.current_app.root_path,
+                           flask.current_app.config["UPLOAD_WEBPAGES"],
+                           newfilename + '.md')
+    if os.path.exists(oldpath) and not os.path.exists(newfilename):
+        os.rename(oldpath, newpath)
+
+
+def read_markdown(name):
+    '''
+    Reads in the mark down text from a file.
+    '''
+
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    curFile = read_file(os.path.join(path, name + '.md'))
+    return curFile
+
+
+def read_file(path):
+    '''
+    Reads in a file
+    '''
+    if not os.path.isfile(path):
+        return ''
+
+    with open(path) as f:
+        return f.read()
+
+
+def get_links():
+    '''
+    Get links for all created webpages
+    '''
+    root = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config["UPLOAD_WEBPAGES"])
+    links = glob.glob(root + '/*')
+    results = []
+    for filenames in links:
+        filenames = filenames.replace(root + '/',
+                                      '').replace('.md', '').replace('_', ' ')
+        link = flask.url_for('uploads.display', url=filenames)
+        results.append((link, filenames))
+    return results
+
+
+def remove_link(filename):
+    '''
+    Get rid of matching filenames
+    '''
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    links = glob.glob(path + '/*')
+    for i in links:
+        name = i.replace(path + '/', '').replace('.md', '').replace("_", " ")
+        if filename == name:
+            os.remove(i)
+    query = """DELETE FROM webpage_files_locks WHERE title = %s"""
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, filename)
+
+
+def check_duplicate(filename):
+    """
+    Check to see if there are duplicate file names
+    """
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    links = glob.glob(path + '/*')
+    filename = filename.replace(' ', '_')
+    for i in links:
+        name = i.replace(path + '/', '').replace('.md', '')
+        if filename == name:
+            return True
+    return False
+
+
+def check_title(title):
+    """
+    Makes sure the title is valid,
+    Allows all numbers and characters. Allows ".", "_", "-"
+    """
+    title_res = re.match("^[0-9a-zA-Z.\/_\- ]*$", title)
+    return title_res != None and len(title) < 100
```

Comment:
Consider using an `r` string for the regular expression. Also, can do the check for length first to avoid computing the regex match if not needed

Reply:
re: check the length: I don't think it's necessary since it's only 100 characters

Reply:
I just mean that it's faster to do the length check than the regex check, and you can stop as soon as either fails. So it makes more sense to check the length first because if it is too long, the regex match doesn't need to be computed.